### ISSUE TITLE
Question about events

### ISSUE THREAD

**Issue by pavelnikolov** [Unleash/unleash#323]
I would like to display an [annotation](http://docs.grafana.org/reference/annotations/) on our Grafana dashboards every time a feature flag is toggled. This would allow us to correlate spikes/issues with feature flag changes. Do you have any idea what is the best way to achieve this?

Thanks

**Comment by ivarconr** [comment]
Hi @pavelnikolov, thanks for you question. 

1. All events are available via the [events api](http://unleash.herokuapp.com/api/admin/events), so it should be possible use that API to create annotations.

2. However, Unleash already exposes some [prometheus metrics](http://unleash.herokuapp.com/internal-backstage/prometheus) and this information could possible be something we should expose here? We could for example just add a counter where we count every time a toggle is changed. This would allow us to create annotations based on prometheus data. 

Do you think it makes sense to look at enabled/disabled, or is it sufficient to just count the changes? Changing the strategies for a feature toggle could have significant impact on how many people the toggle is exposed to.

I'm in favour of pursing 2. Internally in the unleash-server code we already uses an "eventBus" to propagate changes, so it should be "easy" to add a counter every time something changes, without cluttering the existing code. 

What do you think?

**Comment by ivarconr** [comment]
I believe https://github.com/Unleash/unleash/commit/5e591c953df03b3a3e240df3daf9308263657ccd should solve 2) and add a counter for every change to a toggle. 

Having this data in Prometheus should make it possible to create an annotation in grafana for every time a feature toggle is updated.

**Comment by ivarconr** [comment]
I have used to following query for creating annotations: 

```
delta(feature_toggle_update_total{toggle="Demo"}[1m]) != bool 0?
``` 

...it seems to do the trick, creating nice annotations every-time the toggle is updated. 

![image](https://user-images.githubusercontent.com/158948/40423783-df0792aa-5e93-11e8-89a1-e616df9987a5.png)

**Comment by pavelnikolov** [comment]
Awesome @ivarconr I'll check it out! Feel free to close this issue.

**Comment by ivarconr** [comment]
It is not mergers jet, but if this solves it I will try to merge it soon and make a new release.

**Comment by pavelnikolov** [comment]
Well, this is exactly what we need! Can't wait for the release : )

**Comment by ivarconr** [comment]
released as `unleash-server:3.0.4`
